modules/tinymce/src/themes/silver/test/ts/browser/editor/contexttoolbar/ContextToolbarIframePositionTest.ts

```diff
@@ -266,4 +276,66 @@ describe('browser.tinymce.themes.silver.editor.contexttoolbar.ContextToolbarIFra
     TinySelections.select(editor, 'img', []);
     await UiFinder.pWaitForVisible('Waiting for toolbar to appear at the top inside content', SugarBody.body(), '.tox-pop.tox-pop--top');
   });
+
+  it('TINY-7192: Toolbar should flip to the opposite position when the selection overlaps', async () => {
+    const editor = hook.editor();
+    editor.setContent(
+      '<table style="width: 100%; border-collapse: collapse;">' +
+      '<tbody>' +
+      Arr.range(10, (i) => `<tr><td>Cell ${i + 1}</td></tr>`).join('') +
+      '</tbody>' +
+      '</table>'
+    );
+    scrollTo(editor, 0, 0);
+
+    // Select the 1st row in the table, then make sure the toolbar appears at the bottom due to the overlap
+    TinySelections.setCursor(editor, [ 0, 0, 0, 0, 0 ], 0);
+    await UiFinder.pWaitForVisible('Waiting for toolbar to appear at the bottom inside content', SugarBody.body(), '.tox-pop.tox-pop--bottom');
+
+    // select the 4th row (middle of the viewport) and ensure it doesn't move
+    TinySelections.setCursor(editor, [ 0, 0, 3, 0, 0 ], 0);
+    await pAssertHasNotFlipped('.tox-pop.tox-pop--bottom');
+
+    // select the 8th row (bottom of the viewport) and ensure it goes back to the top due to the overlap
+    TinySelections.setCursor(editor, [ 0, 0, 7, 0, 0 ], 0);
+    await UiFinder.pWaitForVisible('Waiting for toolbar to appear at the top inside content', SugarBody.body(), '.tox-pop.tox-pop--top');
+
+    // select the 4th row (middle of the viewport) and again ensure it doesn't move
+    TinySelections.setCursor(editor, [ 0, 0, 3, 0, 0 ], 0);
+    await pAssertHasNotFlipped('.tox-pop.tox-pop--top');
+
+    // Select the 1st row again to make sure it flips back to the bottom
+    TinySelections.setCursor(editor, [ 0, 0, 0, 0, 0 ], 0);
+    await UiFinder.pWaitForVisible('Waiting for toolbar to appear back at the bottom inside content', SugarBody.body(), '.tox-pop.tox-pop--bottom');
+  });
+
+  it('TINY-7192: It should not flip when the selection overlaps and the user is scrolling', async () => {
+    const editor = hook.editor();
+    editor.setContent(
+      '<p style="padding-top: 200px"></p>' +
+      '<table style="width: 100%; border-collapse: collapse;">' +
+      '<tbody>' +
+      Arr.range(10, (i) => `<tr style="height: 22px"><td>Cell ${i + 1}</td></tr>`).join('') +
+      '</tbody>' +
+      '</table>'
+    );
+
+    // Select the 1st row in the table, then make sure the toolbar appears above the table
+    scrollTo(editor, 0, 0);
+    TinySelections.setCursor(editor, [ 1, 0, 0, 0, 0 ], 0);
+    await UiFinder.pWaitForVisible('Waiting for toolbar to appear at the top', SugarBody.body(), '.tox-pop.tox-pop--bottom');
```

Comment:
The comment suggests this is above the table but the class is the bottom?

Reply:
Yes, because it's where the arrow points. So when it appears above the table, the arrow is pointing downwards.